reactor-netty-http/src/main/java/reactor/netty/http/client/Http2Pool.java

```diff
@@ -384,10 +395,15 @@ else if (sig.isOnError()) {
 		}
 	}
 
+	// TODO think on better names for this.  Some kind of signal that this only
+	// returns a connection from the
 	@Nullable
 	Slot findConnection(ConcurrentLinkedQueue<Slot> resources) {
 		int resourcesCount = resources.size();
-		while (resourcesCount > 0) {
+		// TODO this is the KEY idea.  I'm curious on reactor-netty's team
+		// thought on this.  Completely wrong place even if we do some name changes,
+		// cleanup, etc?
+		while (resourcesCount > (minConnections == 0 ? 0 : minConnections + 1)) {
```

Comment:
@violetagg This is the crux of my proposal.  Definitely needs cleanup, but what do you think about this being a good place to force the creation of socket connections?  Honestly, I want to look a little higher up for this, but I am curious if you are fundamentally opposed to this:)

Reply:
Looking at [PR](https://github.com/reactor/reactor-netty/pull/2155/files), I now understand why this is not a good place.  Honestly, I should have looked at the calling code more.  I'm still a little confused by `drainLoop`, but I'm slowly getting there:)